From 2753372e2ae9459f5a877cf313d9e3cb431d4f89 Mon Sep 17 00:00:00 2001
From: Stanislav Vlasic <svlasic@gmail.com>
Date: Mon, 6 Mar 2017 04:23:33 +0100
Subject: [PATCH] Fix EGLNativeTypeAmlogic for probing resolutions

I noticed issue with latest official Amlogic kernel releases (3.10.33)
where /sys/class/amhdmitx/amhdmitx0/disp_cap reports also 4k2k resolutions
on S805 boards which results in 4K resolutions showing as available.

That makes 4K resolutions available to be set which is misleading and in the
end user may actually select unsupported resolution.
---
 xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
index 88cd385..4760c78 100644
--- a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
@@ -160,8 +160,11 @@ bool CEGLNativeTypeAmlogic::ProbeResolutions(std::vector<RESOLUTION_INFO> &resol
   RESOLUTION_INFO res;
   for (std::vector<std::string>::const_iterator i = probe_str.begin(); i != probe_str.end(); ++i)
   {
-    if(aml_mode_to_resolution(i->c_str(), &res))
-      resolutions.push_back(res);
+    if (((StringUtils::StartsWith(i->c_str(), "4k2k")) && (aml_support_h264_4k2k() > AML_NO_H264_4K2K)) || !(StringUtils::StartsWith(i->c_str(), "4k2k")))
+    {
+      if (aml_mode_to_resolution(i->c_str(), &res))
+        resolutions.push_back(res);
+    }
   }
   return resolutions.size() > 0;
 
-- 
1.9.1

